library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -2046,22 +2046,37 @@ void libspdm_set_last_spdm_error_struct(void *context, libspdm_error_struct_t *l
 }
 
 /**
- * Initialize an SPDM context.
+ * Initialize an SPDM context, as well as all secured message contexts,
+ * in the specified locations.
  *
- * The size in bytes of the spdm_context can be returned by libspdm_get_context_size.
+ * The size in bytes of the spdm_context can be returned by
+ * libspdm_get_context_size_without_secured_context.
  *
- * @param  spdm_context                  A pointer to the SPDM context.
+ * The size in bytes of a single secured message context can be returned by
+ * libspdm_secured_message_get_context_size.
  *
- * @retval RETURN_SUCCESS       context is initialized.
- * @retval RETURN_DEVICE_ERROR  context initialization failed.
+ * @param  spdm_context          A pointer to the SPDM context.
+ * @param  secured_contexts      An array of pointers, with each entry containing
+ *                               the location of a secured message context.
+ * @param  num_secured_contexts  Number of secured message contexts to initialize.
+ *                               Currently, only LIBSPDM_MAX_SESSION_COUNT is supported.
+ *
+ * @retval RETURN_SUCCESS        Contexts are initialized.
+ * @retval RETURN_DEVICE_ERROR   Context initialization failed.
  */
-libspdm_return_t libspdm_init_context(void *context)
+libspdm_return_t libspdm_init_context_with_secured_context(void *context,
+                                                           void **secured_contexts,
+                                                           size_t num_secured_contexts)
 {
     libspdm_context_t *spdm_context;
-    void *secured_message_context;
-    size_t SecuredMessageContextSize;
     size_t index;
 
+    if (context == NULL || secured_contexts == NULL ||
```

Comment:
These should be `LIBSPDM_ASSERT`s.

Reply:
Thought to keep as runtime check was because this is input from the integrator, rather than the library itself. In the library itself, we can control to ensure with proper configuration, this is never called with invalid args. We can't necessarily guarantee that from integrator.

If we move to `LIBSPDM_ASSERT`, I should clarify in comments the expectations.

Reply:
The asserts allow an Integrator who knows what they're doing to not have to pay a performance or code size penalty. If an Integrator is unsure of their use of the API then they can enable asserts in debug or release.

Reply:
Fair enough, will change

Reply:
Updated